spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/trace/StackdriverTraceAutoConfiguration.java

```diff
@@ -220,8 +226,23 @@ public Sender stackdriverSender(GcpTraceProperties traceProperties,
 	@Bean
 	@ConditionalOnMissingBean
 	public BaggagePropagation.FactoryBuilder baggagePropagationFactoryBuilder() {
-		return BaggagePropagation.newFactoryBuilder(StackdriverTracePropagation.newFactory(
-				B3Propagation.newFactoryBuilder().injectFormat(B3Propagation.Format.MULTI).build()));
+		Propagation.Factory primary = B3Propagation.newFactoryBuilder().injectFormat(B3Propagation.Format.MULTI).build();
+		return BaggagePropagation.newFactoryBuilder(StackdriverTracePropagation.newFactory(primary));
+	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	public CorrelationScopeCustomizer correlationScopeCustomizer() {
```

Comment:
@adriancole - I couldn't get this to work any other way. Before this change, we had the following values in MDC:

![Screenshot from 2020-10-21 11-43-14](https://user-images.githubusercontent.com/13141550/96774743-756c3980-13b4-11eb-8123-67dfee5ffbf4.png)

But afterwards we only had `traceId` and `spanId`, and NOT their `X-B3-TraceId` and `X-B3-SpanId` equivalents. Is this the intended way to do this? I would imagine the `clear()` at the beginning means multiple systems may not be able to coexists, but I'll admit I'm new to tracing/spanning.

The code as is now only puts these two `X-B3-` values in the MDC

Reply:
configuration is decoupled now. here's how to set the old properties, but mind doing that will re-introduce the performance problems this change fixed https://github.com/spring-cloud/spring-cloud-sleuth/wiki/Spring-Cloud-Sleuth-3.0-Migration-Guide#x-b3--mdc-fields-names-are-no-longer-set

Reply:
Can you give a little more context into the performance issue you were seeing?

Reply:
each stage in a reactive flow re-synchronizes MDC, this is amplified by various properties added to that.

The X-B3- MDC property names were deprecated for a long time and are no longer set in the 3.x release. Continuing to set these by default adds the overhead back (MDC GET/SET on every operator)

Reply:
here are the longer story https://github.com/spring-cloud/spring-cloud-sleuth/pull/1525 https://github.com/spring-cloud/spring-cloud-sleuth/pull/1226 https://github.com/spring-cloud/spring-cloud-sleuth/issues/1397

Reply:
@adriancole Let me see if I understand:
1) For propagation, Brave still uses `X-B3-TraceId` and `X-B3-SpanId`
2) but for MDC logging context, Sleuth switched to instrumentation-library-agnostic `traceId` and `spanId`?

Reply:
@elefeint yep you have it right. these topics are decoupled (incidentally, propagation over JMS always used the "b3" single header)

Reply:
Ok - removed this override and will rely on the instrumentation-agnostic values.